src/Compilers/VisualBasic/Portable/Scanner/Scanner.vb

```diff
@@ -1792,11 +1830,8 @@ FullWidthRepeat:
                         IntegerLiteralStart = Here
                         Base = LiteralBase.Hexadecimal
 
-                        While CanGetCharAtOffset(Here)
-                            ch = PeekAheadChar(Here)
-                            If Not IsHexDigit(ch) Then
-                                Exit While
-                            End If
+                        While TryPeekAheadChar(Here,ch)
+                            If Not IsHexDigit(ch) Then Exit While
```

Comment:
While TryPeekAheadChar(Here, ch) AndAlso IsHexDigit(ch)

The condition is that it is a hex digit. +1 on inlining into the condition.